apps/betterangels-backend/clients/schema.py

```diff
@@ -67,60 +65,124 @@ def _format_graphql_error(error: Exception) -> str:
     errors: list = []
 
-    user_name_dict = {
-        f"{name_field}": user_data.get(name_field) for name_field in ["first_name", "last_name", "middle_name"]
-    }
-    user_name_dict["nickname"] = nickname
+    if california_id in [strawberry.UNSET, None, ""]:
+        return errors
 
-    user_name_untouched = all((v is strawberry.UNSET for v in user_name_dict.values()))
-    user_name_cleared = all((v == "" for v in user_name_dict.values()))
+    california_id: str
+    if not re.search(CALIFORNIA_ID_REGEX, california_id):
+        errors.append({"field": "californiaId", "location": None, "errorCode": ErrorCodeEnum.CA_ID_INVALID.name})
 
-    if user and user.has_name and user_name_untouched:
         return errors
 
-    if user_name_cleared or user_name_untouched:
-        errors.append(_build_error("nickname", None, ErrorMessageEnum.NO_NAME_PROVIDED.name))
+    exclude_arg = {"user_id": user.pk} if user else {}
+
+    if ClientProfile.objects.exclude(**exclude_arg).filter(california_id__iexact=california_id).exists():
+        errors.append({"field": "californiaId", "location": None, "errorCode": ErrorCodeEnum.CA_ID_IN_USE.name})
 
     return errors
 
 
-def _validate_phone_numbers(phone_numbers: list[dict[str, Any]]) -> list[dict[str, Any]]:
+def validate_phone_numbers(phone_numbers: list[dict[str, Any]]) -> list[dict[str, Any]]:
     errors = []
 
     for idx, phone_number in enumerate(phone_numbers):
+        if not phone_number.get("number"):
+            continue
+
         try:
-            phonenumber_field.validators.validate_international_phonenumber(phone_number["number"])
+            validate_international_phonenumber(phone_number["number"])
         except ValidationError:
-            errors.append(_build_error("phoneNumbers", f"{idx}__number", ErrorMessageEnum.INVALID_PHONE_NUMBER.name))
+            errors.append(
+                {
+                    "field": "phoneNumbers",
+                    "location": f"{idx}__number",
+                    "errorCode": ErrorCodeEnum.PHONE_NUMBER_INVALID.name,
+                }
+            )
 
     return errors
 
 
-def _validate_hmis_profiles(hmis_profiles: list[dict[str, Any]]) -> list[dict[str, Any]]:
+def validate_hmis_profiles(hmis_profiles: list[dict[str, Any]]) -> list[dict[str, Any]]:
     errors = []
 
     for idx, hmis_profile in enumerate(hmis_profiles):
-        hmis_profile_id = {"id": hmis_profile["id"]} if hmis_profile.get("id") is not strawberry.UNSET else {}
+        hmis_id = hmis_profile.get("hmis_id")
+
+        if not value_is_set(hmis_id):
+            errors.append(
+                {
+                    "field": "hmisProfiles",
+                    "location": f"{idx}__hmisId",
+                    "errorCode": ErrorCodeEnum.HMIS_ID_NOT_PROVIDED.name,
+                }
+            )
+
+            continue
+
+        hmis_profile_id = {"id": hmis_profile["id"]} if hmis_profile.get("id") else {}
```

Comment:
for consistency:
```suggestion
        # exclude id of already persisted hmis profile to avoid incorrectly flagging duplicate 
        exclude_arg = {"id": hmis_profile["id"]} if hmis_profile.get("id") else {}
```
or something like that